src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -94,11 +95,23 @@ public function getProxyFactoryCode(Definition $definition, $id, $factoryCode =
      */
     public function getProxyCode(Definition $definition)
     {
-        return preg_replace(
+        $code = $this->classGenerator->generate($this->generateProxyClass($definition));
+
+        $code = preg_replace(
             '/(\$this->initializer[0-9a-f]++) && \1->__invoke\(\$this->(valueHolder[0-9a-f]++), (.*?), \1\);/',
             '$1 && ($1->__invoke(\$$2, $3, $1) || 1) && $this->$2 = \$$2;',
-            $this->classGenerator->generate($this->generateProxyClass($definition))
+            $code
         );
+
+        if (version_compare(Version::getVersion(), '2.2', '<')) {
+            $code = preg_replace(
+                '/((?:\$(?:this|initializer|instance)->)?(?:publicProperties|initializer|valueHolder))[0-9a-f]+/',
```

Comment:
should use a possessive quantifier to avoid backtracking (as done in the previous regex)

Reply:
I had no idea what possessive quantifiers are but I hope I got what it means